### PR TITLE
add gsets support

### PR DESCRIPTION
**Pull request by binarytemple-bet365**

### CODE REVIEW

File: src/riak_dt.proto [hunk 1]
```diff
@@ -154,6 +156,18 @@ message SetOp {
...
+
+/*
+ * An operation to update a GSet, on its own.
+ * GSet members are opaque binary values, you can only add
+ * them to a Set.
+ */
+message GSetOp {
+    repeated bytes adds    = 1;
+}
+
```
Comment (lukebakken) [inline src/riak_dt.proto]: Whitespace

File: src/riak_dt.proto [hunk 2]
```diff
@@ -107,6 +107,7 @@ message DtValue {
     optional sint64   counter_value = 1;
     repeated bytes    set_value     = 2;
     repeated MapEntry map_value     = 3;
+    repeated bytes    gset_value    = 4;
```
Comment (lukebakken) [inline src/riak_dt.proto]: Unnecessary with `DataType` discriminator.

File: src/riak_dt.proto [hunk 3]
```diff
@@ -251,4 +267,6 @@ message DtUpdateResp {
     optional sint64   counter_value = 3;
     repeated bytes    set_value     = 4;
     repeated MapEntry map_value     = 5;
-}
\ No newline at end of file
+    repeated bytes    gset_value    = 6;
```
Comment (lukebakken) [inline src/riak_dt.proto]: I will argue that this is also unnecessary. When making an update, the sender knows if they sent a `SetOp` or `GSetOp`. `set_value` is the only field needed.

File: src/riak.proto [hunk 4]
```diff
@@ -9,7 +9,7 @@
 ** except in compliance with the License.  You may obtain
 ** a copy of the License at
 **
-**   http://www.apache.org/licenses/LICENSE-2.0
+**   http://www.apache.org/licenses/LICENSE-1.0
```
Comment (russelldb) [inline src/riak.proto]: Is this license change necessary? Intentional?

File: rebar.config [hunk 5]
```diff
@@ -1,12 +1,12 @@
 %% -*- mode: erlang -*-
 {erl_opts, [
     debug_info,
-    warnings_as_errors,
     {platform_define, "^[0-9]+", namespaced_types}
 ]}.
 
 {deps, [
     {protobuffs, "0.8.*", {git, "https://github.com/basho/erlang_protobuffs.git", {tag, "0.8.4"}}},
+    {meck, "0.8.2", {git, "git://github.com/basho/meck.git", {tag, "0.8.2"}}},
```
Comment (binarytemple-bet365) [inline rebar.config]: may remove before end of day - haven't needed it yet.

File: rebar.config [hunk 6]
```diff
@@ -1,7 +1,6 @@
 %% -*- mode: erlang -*-
 {erl_opts, [
     debug_info,
-    warnings_as_errors,
```
Comment (binarytemple-bet365) [inline rebar.config]: temporary, cutting and chopping code

File: src/riak_dt.proto (hunk 1 above)
Comment (binarytemple-bet365) [inline src/riak_dt.proto]: fixed

**Comment by binarytemple-bet365** [comment]
@lukebakken discussed with @russelldb , this approach isn't working out, closing this P/R - will review and reopen my original design when I get some free time

**Pull request closed without merging**